### DISCUSSION TITLE
Rustdesk and Citrix (rdp session sharing)

### DISCUSSION THREAD

**Discussion by steverileysyfan** [rustdesk/rustdesk#937] [Q&A]
Hello, we have implemented RustDesk on our environment, and it works great, except with users that are on a citrix farm.  when connecting to a user that is on a citrix farm, the remote connection comes up as a RDP session that you have to log into, even though you are connecting to an ID that exists for a specific user on the citrix Server farm.  I have searched the internet and github, but cannot find anything pertaining to RustDesk working on Citrix Farms.  Thank you in advance for you assistance on getting this working, or knowing what we are doing wrong to allow us to connect through rustdesk to our users on Citrix Farm.

**Comment by rustdesk** [comment]
The problem is we do not have Citrix Farm, even never know it.

**Comment by rustdesk** [comment]
Why do you want to use RustDesk in the Farm? As far as I know, Citrix has full remote desktop support, though I have never used Citrix products (seems no personal free edition), but I know Citrix developed RDP.

**Reply by steverileysyfan** [reply]
we use Citrix farm as a Thin client environment, and want to be able to use RustDesk to easily connect to a users desktop(instance) so we can see exactly what they see.  using RDP gives us a separate desktop environment from what the user is seeing at that moment, so we cannot see the issue the user is dealing with.

**Reply by rustdesk** [reply]
So Citrix's all products are built on RDP?

**Reply by rustdesk** [reply]
We have RDP session support, https://github.com/rustdesk/rustdesk/issues/914#issuecomment-1173839858

**Reply by steverileysyfan** [reply]
yes Citrix uses RDP for their Desktop environment

**Reply by steverileysyfan** [reply]
I will check out that comment, thank you

**Reply by rustdesk** [reply]
> yes Citrix uses RDP for their Desktop environment

Does it support Mac/Linux?

**Reply by steverileysyfan** [reply]
We have Citrix setup running windows server 2019 as the desktop environment.  I believe Citrix would allow Mac and Linux to be used as host desktops

**Reply by spencercolletsyfan** [reply]
I believe Citrix uses ICA instead of RDP although RDP is enabled on the servers themselves. RDP session sharing does not fix this issue.

**Reply by rustdesk** [reply]
> ICA

Sorry we can not help you, we have no chance to expeirence ICA.

**Reply by rustdesk** [reply]
> ICA

Can you run `query session` in cmd, and give me the output? I wanna know how to detect ICA session.

**Reply by agc-gthub** [reply]
Good morning, I have seen this topic and I add information to add support for Citrix connections (ICA protocol).
![image](https://github.com/user-attachments/assets/03216347-2cdb-4a35-982d-fce54096d95a)

There are two types :
ica-cgp
ica-tcp


Thank you.
Regards.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/commit/7305b6bd1c815429dbcdd6c8dd57a5ca16b01375 I add some code for this, you can  test tomorrow when the new nightly build is ready. This affects controlled side.

**Reply by agc-gthub** [reply]
Perfect, thank you very much. I will install the new version tomorrow and confirm if everything works correctly.

Best regards.

**Reply by agc-gthub** [reply]
Good afternoon, I have tested the client with the changes and it already lists the ICA sessions inside the server. Thank you very much.
